### PR TITLE
Updated backend README.md with the latest Cassandra schemas

### PR DESCRIPTION
**Pull request by Shoukozumi**
Updated Clio's back end README.md with the latest Cassandra schemas and diagram-aided explanations of the successor table.

### CODE REVIEW

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]:
```suggestion
The backend of Clio is responsible for handling the proper reading and writing of past ledger data from and to a given database. As of right now, Cassandra is the only supported database that is production-ready. However, support for more databases like PostgreSQL and DynamoDB may be added in future versions. Support for database types can be easily be extended by creating new implementations which implements the virtual methods of `BackendInterface.h`. Then, use the Factory Object Design Pattern to simply add logic statements to `BackendFactory.h` that return the new database interface for a specific `type` in Clio's configuration file. 
```

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]:
```suggestion
The data model used by Clio to read and write ledger data is different than what Rippled uses. Rippled uses a novel data structure named [*SHAMap*](https://github.com/ripple/rippled/blob/master/src/ripple/shamap/README.md), which is a combination of a Merkle Tree and a Radix Trie. In a SHAMap, ledger objects are stored in the root vertices of the tree. Thus, looking up a record located at the leaf node of the SHAMap executes a tree search, where the path from the root node to the leaf node is the key of the record. Rippled nodes can also generate a proof-tree by forming a subtree with all the path nodes and their neighbors, which can then be used to prove the existnce of the leaf node data to other Rippled nodes. In short, the main purpose of the SHAMap data structure is to facilitate the fast validation of data integrity between different decentralized Rippled nodes.
```

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]:
```suggestion
There are three main types of data in each XRP ledger version, they are [Ledger Header](https://xrpl.org/ledger-header.html), [Transaction Set](https://xrpl.org/transaction-formats.html) and [State Data](https://xrpl.org/ledger-object-types.html). Due to the structural differences of the different types of databases, Clio may choose to represent these data using a different schema for each unique database type. 
```

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]: Call this *Ledger Entry* or *Ledger Object* or "Ledger State", not just *Data*

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]:
```suggestion
In Cassandra, Clio will be creating 9 tables to store the ledger data, they are `ledger_transactions`, `transactions`, `ledger_hashes`, `ledger_range`, `objects`, `ledgers`, `diff`, `account_tx`, and `successor`.  Their schemas and how they work are detailed below.
```
These are called *tables* keyspace is more similar to a "database" in PostgreSQL.

File: src/backend/README.md
Comment (natenichols) [inline src/backend/README.md]: We may want to add this .png file to display in the .md file, instead of just linking it.

**Review by natenichols — APPROVED** [review]
Great work. I have a couple of comments, but the information contained looks correct to me.

Lets let CJ review it too, before we merged, but I have no comments that should hold up approval.

File: src/backend/README.md
Comment (cjcobb23) [inline src/backend/README.md]: I'd change to "This table stores the hashes of all transactions in a given ledger ..."

File: src/backend/README.md
Comment (cjcobb23) [inline src/backend/README.md]: I'd change to "this table stores the full transaction and metadata of each validated transaction ..."

File: src/backend/README.md
Comment (cjcobb23) [inline src/backend/README.md]: Let's add a paragraph here that says something like "This table is updated when all data for a given ledger sequence has been written to the various tables in the database. For each ledger, many associated records are written to different tables. This table is used as a synchronization mechanism, to prevent the application from reading data from a ledger for which all data has not yet been fully written."

File: src/backend/README.md
Comment (cjcobb23) [inline src/backend/README.md]: This table stores the list of transactions affecting a given account. This includes transactions made by the account, as well as transactions received.

File: src/backend/README.md
Comment (cjcobb23) [inline src/backend/README.md]: The example query needs an equality operation on the key `SELECT * FROM successor WHERE key = ? and seq <= n ORDER BY seq DESC LIMIT 1`

Comment (Shoukozumi) [reply src/backend/README.md]: Fixed

File: src/backend/README.md
Comment (Shoukozumi) [reply src/backend/README.md]: Fixed

File: src/backend/README.md
Comment (Shoukozumi) [reply src/backend/README.md]: Fixed

File: src/backend/README.md
Comment (Shoukozumi) [reply src/backend/README.md]: Fixed

File: src/backend/README.md
Comment (Shoukozumi) [reply src/backend/README.md]: Fixed

File: src/backend/README.md
Comment (cjcobb23) [reply src/backend/README.md]: I think you might have missed this one. It still says ".. of each ledger version" but should say "..of each validated transaction"

**Review by cjcobb23 — APPROVED** [review]
LGTM

**Pull request merged**
